spring-boot-project/spring-boot/src/main/java/org/springframework/boot/rsocket/context/RSocketPortInfoApplicationContextInitializer.java

```diff
@@ -51,49 +51,43 @@ public void initialize(ConfigurableApplicationContext applicationContext) {
 		applicationContext.addApplicationListener(new Listener(applicationContext));
 	}
 
-	private static class Listener implements ApplicationListener<RSocketServerInitializedEvent> {
+	private record Listener(ConfigurableApplicationContext applicationContext) implements ApplicationListener<RSocketServerInitializedEvent> {
```

Comment:
Making this a `record` adds an accessor for the application context which we don't want. Please revert.